geode-wan/src/main/java/org/apache/geode/cache/client/internal/locator/wan/LocatorMembershipListenerImpl.java

```diff
@@ -328,6 +331,7 @@ public void run() {
                   Thread.currentThread().interrupt();
                   logger.warn(
                       "Locator Membership listener permanently failed to exchange locator information due to interruption.");
+                  return;
```

Comment:
If a sleep is interrupted we now bail out of `run()` entirely, rather than continuing on to attempt delivering the message to the next target. Why the change?

Reply:
The exception handler was leaving the interrupt bit set.  Nothing useful could happen after that.  Any I/O operations that it attempted would fail, resulting in the thread hanging around sleeping until it gave up;.